shanoir-ng-datasets/src/main/java/org/shanoir/ng/examination/repository/ExaminationRepository.java

```diff
@@ -61,17 +62,30 @@ public interface ExaminationRepository extends PagingAndSortingRepository<Examin
 	 */
 	List<Examination> findByStudyIdIn(List<Long> studyIds);
 
+	@Query(value = "SELECT * FROM examination e WHERE e.study_id in #studyIds "
+			+ "JOIN study_user_center suc "
+			+ "ON suc.center_id = e.center_id"
+			+ "AND suc.study_id = e.center_id;",
+			    countQuery = "SELECT count(*) FROM examination e WHERE e.study_id in #studyIds "
+						+ "JOIN study_user_center suc "
+						+ "ON suc.center_id = e.center_id"
+						+ "AND suc.study_id = e.center_id;",
+			    nativeQuery = true)
+	Page<Examination>findByStudyIdInFilterByCenter(List<Long> studyIds, Pageable pageable);
+	
 	/**
 	 * Get a paginated list of examinations for a list of studies.
 	 * 
 	 * @param studyIds
 	 *            list of study ids.
-	 * @param pageable
+	 * @param sort
 	 *            pagination data.
 	 * @return list of examinations.
 	 */
 	Page<Examination> findByPreclinicalAndStudyIdIn(Boolean preclinical, List<Long> studyIds, Pageable pageable);
 
+	List<Examination> findByPreclinicalAndStudyIdIn(Boolean preclinical, List<Long> studyIds, Sort sort);
```

Comment:
could 

`Page<Examination>findByStudyIdInFilterByCenter(List<Long> studyIds, Pageable pageable);`

replace both 

`Page<Examination> findByPreclinicalAndStudyIdIn(Boolean preclinical, List<Long> studyIds, Pageable pageable);`

and

`List<Examination> findByPreclinicalAndStudyIdIn(Boolean preclinical, List<Long> studyIds, Sort sort);`

if we add a preclinical option to it ?

Reply:
What does "FilterByCenter" part is doing ? Maybe we should not mix Page result and List results